### PR TITLE
add test case for Empty() and ToQueue()

### PR DESCRIPTION
**Pull request by prangnt**
add test case for Empty() and ToQueue()

### CODE REVIEW

File: shared/datatype/queue.go [hunk 1]
```diff
@@ -28,15 +28,26 @@ func (q *Queue) Enqueue(d interface{}) *Queue {
 func (q *Queue) Get() interface{} {
 	var data = q.Head()
 
-	// Remove data from first queue
-	q.internal[0] = nil // free memory
-	q.internal = q.internal[1:]
+	if data == nil {
```
Comment (kamontat) [inline shared/datatype/queue.go:31]: This `if` is unnecessary.

File: shared/datatype/queue.go [hunk 2]
```diff
@@ -28,15 +28,26 @@ func (q *Queue) Enqueue(d interface{}) *Queue {
...
+		q.internal[0] = nil // free memory
+		q.internal = q.internal[1:]
 
+	}
 	return data
 }
 
 // Head will get first queue but not change queue size
 func (q *Queue) Head() interface{} {
+
```
Comment (kamontat) [inline shared/datatype/queue.go:46]: could you remove this space before if condition.

File: shared/datatype/queue_test.go [hunk 3]
```diff
@@ -50,4 +50,43 @@ func TestNormalQueue(t *testing.T) {
...
+		WithExpected(nil).
+		WithActual(queue1.Get()).
+		MustEqual()
+
+	assertion.NewName("should return empty").
+		WithExpected(true).
+		WithActual(datatype.NewLimitQueue(0).Empty()).
+		MustEqual()
+
+	assertion.NewName("should return empty").
```
Comment (kamontat) [inline shared/datatype/queue_test.go:81]: Is your test message wrong?

**Review by kamontat — CHANGES_REQUESTED** [review]

**Review by kamontat — APPROVED** [review]
